storage/blob.go

```diff
@@ -629,7 +630,13 @@ func (b BlobStorageClient) CreateBlockBlob(container, name string) error {
 func (b BlobStorageClient) CreateBlockBlobFromReader(container, name string, size uint64, blob io.Reader) error {
 	path := fmt.Sprintf("%s/%s", container, name)
 	uri := b.client.getEndpoint(blobServiceName, path, url.Values{})
+	splitted := strings.Split(name, ".")
+	if len(splitted) == 0 {
+		splitted = []string{""}
```

Comment:
instead of this empty slice hack can you please do:

    if len(splitted) > 1 { headers["Content-Type"] = mime.TypeByExtension("." + splitted[len(splitted)-1])  }